lib/provisioner/worker/plugins/automators/chef_solo_automator/chef_solo_automator.rb

```diff
@@ -51,13 +53,44 @@ def generate_chef_primitive_tar(chef_primitive)
     # rubocop:disable GuardClause
     if !File.exist?(chef_primitive_tar) || ((Time.now - File.stat(chef_primitive_tar).mtime).to_i > 600)
       log.debug "Generating #{chef_primitive_tar} from #{chef_primitive_path}"
-      `tar -chzf "#{chef_primitive_tar}.new" #{chef_primitive}`
+      newfile = gzip(tar(chef_primitive_path))
+      File.new("#{chef_primitive_tar}.new", 'w').write newfile.read
+      # to do: add a test for tarball creation (if !File.exist?("#{chef_primitive_tar}.new") => rescue?)
       `mv "#{chef_primitive_tar}.new" "#{chef_primitive_tar}"`
       log.debug "Generation complete: #{chef_primitive_tar}"
     end
     # rubocop:enable GuardClause
   end
 
+  def tar(path)
```

Comment:
can you copy the same comments describing each of these methods from data-uploader?